plugin-gradle/src/test/java/com/diffplug/gradle/spotless/KotlinExtensionTest.java

```diff
@@ -40,4 +44,29 @@ public void integration() throws IOException {
 		String formatted = getTestResource("kotlin/ktlint/basic.clean");
 		Assert.assertEquals(formatted, result);
 	}
+
+	@Test
+	public void testWithHeader() throws IOException {
+		write("build.gradle",
+				"plugins {",
+				"    id 'nebula.kotlin' version '1.0.6'",
+				"    id 'com.diffplug.gradle.spotless'",
+				"}",
+				"repositories { mavenCentral() }",
+				"spotless {",
+				"    kotlin {",
+				"        licenseHeader('" + HEADER + "')",
+				"        ktlint()",
+				"    }",
+				"}");
+		final File testFile = write("src/main/kotlin/test.kt", getTestResource("kotlin/licenseheader/KotlinCodeWithoutHeader.test"));
+		final String original = read(testFile.toPath());
+		gradleRunner().withArguments("spotlessApply").build();
+		final String result = read(testFile.toPath());
+		// Make sure the header gets added.
+		Assertions.assertThat(result).startsWith(HEADER);
+		// Make sure that the rest of the file is still there with nothing removed.
+		Assertions.assertThat(result).endsWith(original);
```

Comment:
AFAICT, these two assertions don't guard against the case where unrelated text gets inserted in between the header and the rest of the original file contents, accidentally or otherwise.

Thus I wonder if it's worth testing that `original` directly follows `HEADER`.

(If so, one might be able to do this using String character indexes somehow, or alternatively checking that the file contains exactly as many lines as `HEADER + original`...)

Reply:
Interesting thought. I'll add something to assert that.

Reply:
I've expanded this test.